azure_monitor/src/azure_monitor/sdk/auto_collection/metrics_span_processor.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import logging
+
+from opentelemetry.trace import SpanKind
+from opentelemetry.sdk.trace import Span, SpanProcessor
+
+from azure_monitor.export import convert_span_to_envelope
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMetricsSpanProcessor(SpanProcessor):
+    """AzureMetricsSpanProcessor is an implementation of `SpanProcessor` used
+    to generate Azure specific metrics, including dependencies/requests rate, average duration
+    and failed dependencies/requests.
+    """
+
+    def __init__(self, exporter=None):
+        self._exporter = exporter
+        self.request_count = 0
+        self.dependency_count = 0
+        self.failed_request_count = 0
+        self.failed_dependency_count = 0
+        self.request_duration = 0
+        self.dependency_duration = 0
+
+    def on_start(self, span: Span) -> None:
+        pass
+
+    def on_end(self, span: Span) -> None:
+        try:
+            if span.kind == SpanKind.SERVER:
+                self.request_count = self.request_count + 1
```

Comment:
I think there might be an issue with using the spanprocessor on_end as the indicator to request/dependency happening. First of all, the counts/durations would be inconsistent depending on how long it takes the span to finish. Secondly, this requires the user to be instrumented with the required integrations for this to work (spans have to be generated). This might actually be a desired design to have but currently other SDK (Opencensus) does not behave like this. Might be good to discuss.